src/components/OrderBookWidget.tsx

```diff
@@ -212,115 +131,84 @@ const processData = (
   return points
 }
 
-function _printOrderBook(pricePoints: PricePointDetails[], baseToken: TokenDetails, quoteToken: TokenDetails): void {
-  logDebug('Order Book: ' + baseToken.symbol + '-' + quoteToken.symbol)
+function _printOrderBook(pricePoints: PricePointDetails[], baseTokenSymbol = '', quoteTokenSymbol = ''): void {
+  logDebug('Order Book: ' + baseTokenSymbol + '-' + quoteTokenSymbol)
   pricePoints.forEach((pricePoint) => {
     const isBid = pricePoint.type === Offer.Bid
     logDebug(
-      `\t${isBid ? 'Bid' : 'Ask'} ${pricePoint.totalVolumeFormatted} ${baseToken.symbol} at ${
+      `\t${isBid ? 'Bid' : 'Ask'} ${pricePoint.totalVolumeFormatted} ${baseTokenSymbol} at ${
         pricePoint.priceFormatted
-      } ${quoteToken.symbol}`,
+      } ${quoteTokenSymbol}`,
     )
   })
 }
 
-const draw = (
-  chartElement: HTMLElement,
-  baseToken: TokenDetails,
-  quoteToken: TokenDetails,
-  networkId: number,
-  hops?: number,
-): am4charts.XYChart => {
-  const baseTokenLabel = safeTokenName(baseToken)
-  const quoteTokenLabel = safeTokenName(quoteToken)
-  const market = baseTokenLabel + '-' + quoteTokenLabel
-  am4core.useTheme(am4themesSpiritedaway)
-  am4core.options.autoSetClassName = true
-  const chart = am4core.create(chartElement, am4charts.XYChart)
-  const networkDescription = networkId !== Network.Mainnet ? `${getNetworkFromId(networkId)} ` : ''
-
-  // Add data
-  chart.dataSource.url = dexPriceEstimatorApi.getOrderBookUrl({
-    baseTokenId: baseToken.id,
-    quoteTokenId: quoteToken.id,
-    hops,
-    networkId,
-  })
-  chart.dataSource.adapter.add('parsedData', (data) => {
-    try {
-      const bids = processData(data.bids, baseToken, quoteToken, Offer.Bid)
-      const asks = processData(data.asks, baseToken, quoteToken, Offer.Ask)
-      const pricePoints = bids.concat(asks)
+interface ProcessRawDataParams {
+  data: RawApiData
+  baseToken: Pick<TokenDetails, 'decimals' | 'symbol'>
+  quoteToken: Pick<TokenDetails, 'decimals' | 'symbol'>
+}
 
-      // Sort points by price
-      pricePoints.sort((lhs, rhs) => lhs.price.comparedTo(rhs.price))
+export const processRawApiData = ({ data, baseToken, quoteToken }: ProcessRawDataParams): PricePointDetails[] => {
```

Comment:
Maybe we should promote methods like this one to a util, like `src/utils/orderbook.ts`
With that, amazing if we add some very basic test too.